### PR TITLE
Client and target request authorization with separate scopes and request management for the belonging offers or target systems by it's client id

### LINKED ISSUE #27: The client can only manage it's own offers, but not from a different client
* extract the client id from the access token in the request using Keycloak
* save it with the offer
* evaluate client/target requests by client id

### LINKED ISSUE #29: Client and target system use different scopes for request authorization
Currently the scope is the same.

### PR DESCRIPTION
**Pull request by kat-ka**
* get client id from request access token
* save it with offers and target systems
* evaluate requests by offers and target systems belonging to this client id
* use different scopes for client and target requests
* test cases are included

### CODE REVIEW

File: src/main/java/org/sharesquare/hub/configuration/SecurityConfiguration.java [hunk 1]
```diff
@@ -49,8 +53,11 @@ JwtAuthenticationConverter createJwtAuthenticationConverter() {
     @Value("${SHARE2_USER_ID_CLAIM:user_id}")
     private String userIdClaim;
 
-	@Value("${custom.auth.server.scope}")
-	private String scope;
+	@Value("${custom.auth.server.scope.offers}")
```
Comment (yncyrydybyl) [inline src/main/java/org/sharesquare/hub/configuration/SecurityConfiguration.java]: crazy indentation

File: src/main/java/org/sharesquare/hub/configuration/SecurityConfiguration.java [hunk 2]
```diff
@@ -97,7 +104,10 @@ protected void configure(HttpSecurity http) throws Exception {
 				.mvcMatchers("/v3/api-docs/**", "/swagger-ui.html", "/swagger-ui/**").permitAll() // Swagger UI
 				.antMatchers("/h2-console/**").permitAll() // H2 Web Console
 				.antMatchers("/actuator/**").permitAll()
-				.mvcMatchers("/offers/**", "/targetSystems/**").hasAuthority(SCOPE_PREFIX + scope)
+				.mvcMatchers("/offers/**").hasAuthority(SCOPE_PREFIX + offersScope)
+				.mvcMatchers(GET, "/targetsystems/**").hasAuthority(SCOPE_PREFIX + offersScope)
```
Comment (yncyrydybyl) [inline src/main/java/org/sharesquare/hub/configuration/SecurityConfiguration.java]: /targetsystems with offersScope ? Thats a typo I guess

Comment (kat-ka) [reply src/main/java/org/sharesquare/hub/configuration/SecurityConfiguration.java]: As I understood it, the client with the offers scope is allowed to get a list of the available target systems he can choose from. And the target system is only allowed to create or delete a target system entry.

File: src/main/java/org/sharesquare/hub/configuration/SecurityConfiguration.java (hunk 1 above)
Comment (kat-ka) [reply src/main/java/org/sharesquare/hub/configuration/SecurityConfiguration.java]: Changed it.

**Review by yncyrydybyl — APPROVED** [review]

**Pull request merged**